### PR TITLE
Fix building for iOS and macOS

### PR DESCRIPTION
**Pull request by yorickreum**
This PR fixes building of XCSoar for iOS and macOS (ARM64).

For iOS, the showstopper was the incomplete implementation of SDL. 
For macOS, FMT was not included correctly (compare https://github.com/yorickreum/XCSoar/actions/runs/10444645053/job/28919419280 - I'm not sure why it works for other platforms without these depends).

Major shoutout to @WhyToFly and @piopawlu for all their debugging and ideas!

Given the XCSoar community is still interested in this, I'm happy to provide more pull requests with bug fixes and improvements on the Apple support of XSoar in the next days.

### CODE REVIEW

**Comment by piopawlu** [comment]
Hey! Great that there is interest and effort put to get XCSoar to build and work on iOS again :-) Did you check if it actually works and renders correctly on an iOS device? From my work I remember that getting it to build was one thing but then it wouldn't render the map or its controls att all. I had to make changes in src/ui/canvas/opengl/RenderBuffer.hpp, src/ui/canvas/opengl/FrameBuffer.hpp and only after that it started working.

**Comment by yorickreum** [comment]
> Hey! Great that there is interest and effort put to get XCSoar to build and work on iOS again :-) Did you check if it actually works and renders correctly on an iOS device? 

Hey @piopawlu! You are absolutely right that with only this PR it won’t yet render correctly on iOS. I did plan on also committing the fixes on the OpenGL implementation that you gratefully figured out, but I planned on doing so in a separate pull request to keep things clean (and also to only put the work in once I’m sure that there actually is interest in keeping this official repository iOS compatible).

**Comment by piopawlu** [comment]
Understood! And thanks for continuing this work 👍

File: .github/workflows/build-native.yml
Comment (lordfolken) [inline .github/workflows/build-native.yml:246]: Please create a shell script for osx macos in ide/provisioning/ analog install-debian-packages.sh

The dependency management should be not in github actions, so we depend as little as possible on it.

File: .github/workflows/build-native.yml
Comment (lordfolken) [inline .github/workflows/build-native.yml:156]: Do we need to keep macos-12? i'm fine to discard it if not needed.

**Review by lordfolken — CHANGES_REQUESTED** [review]
Thanks for this. Are you willing to maintain IOS/OSX ports long term? 

Please do your git commit messages in the format which we use:

file/area: change description
additional notes etc

Please no fixup commits, such as the last one. 

git rebase -i HEAD~2 to edit the git history. 
git push --force to force-push.

File: .github/workflows/build-native.yml
Comment (lordfolken) [reply .github/workflows/build-native.yml]: Perfect thx.

File: .github/workflows/build-native.yml
Comment (yorickreum) [reply .github/workflows/build-native.yml:156]: I kept this at the macOS 12 GitHub actions container to still allow building for x86-64 without the need to cross-compile from an ARM architecture. I do think it's good to still be able to build for x64, given that a lot of macOS devices in use are still not ARM based yet.

**Review by lordfolken — CHANGES_REQUESTED** [review]
Very good thx. One small thing:

Please add a newline at the end of the shell script and chmod 0755 it to set executable bits.

File: ide/provisioning/install-darwin-packages.sh
Comment (lordfolken) [inline ide/provisioning/install-darwin-packages.sh:1]: can you make it either bash or sh? zsh is great, but not natively installed in many places.

Comment (lordfolken) [reply ide/provisioning/install-darwin-packages.sh:1]: Or is it on osx?

File: ide/provisioning/install-darwin-packages.sh
Comment (lordfolken) [inline ide/provisioning/install-darwin-packages.sh:63]: New line missing at end of file.

File: .github/workflows/build-native.yml
Comment (lordfolken) [reply .github/workflows/build-native.yml:156]: Good Point. 
Currently this one fails to compile with:
A```
pplying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/ublas_no_serialization.patch
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_math_throw
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_multiprecision
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/no_iostream
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_no_backward_compatibility
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_math
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/variant_no_type
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/json_no_hash
Applying patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_array
Now at patch /Users/runner/work/XCSoar/XCSoar/lib/boost/patches/geometry_std_array
Package sdl2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'sdl2' found
build/compile.mk:104: *** library not found: sdl2.  Stop.
gmake: 
```

File: ide/provisioning/install-darwin-packages.sh
Comment (yorickreum) [reply ide/provisioning/install-darwin-packages.sh]: Changed it to bash. Indeed, in macOS 12 zsh was not preinstalled (or default shell) yet.

**Comment by yorickreum** [comment]
> [...] and chmod 0755 it to set executable bits.

Not sure, according to ```git ls-tree HEAD``` it's already 755.

**Comment by yorickreum** [comment]
> Thanks for this. Are you willing to maintain IOS/OSX ports long term?

I'll try to.

File: .github/workflows/build-native.yml
Comment (yorickreum) [reply .github/workflows/build-native.yml:156]: Dependencies were not getting installed correctly for OSX, fixed.

**Review by lordfolken — APPROVED** [review]

**Comment by LeNinjaHD** [comment]
I am currently trying to compile XCSoar for iOS, however i get this error:

  File "/opt/homebrew/Cellar/python@3.13/3.13.1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ninja', '-v', 'install']' returned non-zero exit status 1.

Does anyone have some advice?

**Comment by lordfolken** [comment]
Please don't comment on closed PRs. Either raise a new issue or discussion. 
ninja -v install was not successful. Beware I don´t have apple hardware, so I cannot really help you. Maybe @yorickreum

**Comment by LeNinjaHD** [comment]
> Please don't comment on closed PRs. Either raise a new issue or discussion. ninja -v install was not successful. Beware I don´t have apple hardware, so I cannot really help you. Maybe @yorickreum

Sorry, opened a Forum Thread.

**Comment by yorickreum** [comment]
> Sorry, opened a Forum Thread.

Which forum? I'm happy to try helping you with this, but I currently cannot reproduce the issue (for me, all ninja builds work out fine), so I'd need more details on where exactly the build fails. I agree with @lordfolken that we should discuss this elsewhere.

**Comment by LeNinjaHD** [comment]
> > Sorry, opened a Forum Thread.
> 
> Which forum? I'm happy to try helping you with this, but I currently cannot reproduce the issue (for me, all ninja builds work out fine), so I'd need more details on where exactly the build fails. I agree with @lordfolken that we should discuss this elsewhere.

Its still waiting for approval. How can i contact you?

**Comment by yorickreum** [comment]
> > > Sorry, opened a Forum Thread.
> > 
> > 
> > Which forum? I'm happy to try helping you with this, but I currently cannot reproduce the issue (for me, all ninja builds work out fine), so I'd need more details on where exactly the build fails. I agree with @lordfolken that we should discuss this elsewhere.
> 
> Its still waiting for approval. How can i contact you?

I'd suggest that you open an issue here https://github.com/XCSoar/XCSoar/issues -- you can tag me there and we can just discuss there.

**Comment by LeNinjaHD** [comment]
> > > > Sorry, opened a Forum Thread.
> > > 
> > > 
> > > Which forum? I'm happy to try helping you with this, but I currently cannot reproduce the issue (for me, all ninja builds work out fine), so I'd need more details on where exactly the build fails. I agree with @lordfolken that we should discuss this elsewhere.
> > 
> > 
> > Its still waiting for approval. How can i contact you?
> 
> I'd suggest that you open an issue here https://github.com/XCSoar/XCSoar/issues -- you can tag me there and we can just discuss there.

#1625

**Pull request merged**
